src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -289,21 +289,17 @@ void primitive_inst::update_impl() {
         };
 
         auto layout_key = get_layout_key();
-        auto& cache = _network.get_program()->get_implementations_cache();
+        auto& cache = get_network().get_program()->get_implementations_cache();
         if (cache.has(layout_key)) {
             _impl = cache.get(layout_key)->clone();
             GPU_DEBUG_PROFILED_STAGE_CACHE_HIT(true);
         } else {
-            auto lru = cache.get_lru_element();
             _impl = _node.type()->choose_impl(_node, *_impl_params);
-            bool lru_popped = cache.add(layout_key, _impl->clone());
-            if (lru_popped) {
-                for (auto& id : lru->get_kernel_ids())
-                    _network.get_program()->remove_kernel(id);
-            }
-            _network.get_program()->compile();
+            _impl->add_kernel_codes(get_network().get_kernels_cache());
+            get_network().compile();
+            _impl->init_kernels(get_network().get_kernels_cache());
```

Comment:
I'd change the interfaces to something like that:
```cpp
auto& kernels_cache = get_network().get_kernels_cache();
kernels_cache.add(_impl->get_kernels_data());
kernels_cache.compile();
_impl->init_kernels(kernels_cache);
kernels_cache.reset();
```
or even that:
```cpp
auto& kernels_cache = get_network().get_kernels_cache();
auto kernels = kernels_cache.compile_kernels(_impl->get_kernels_data());
_impl->set_kernels(kernels);
```

Reply:
@vladimir-paramuzov I think the second is better. When do you merge this code change to master?

Reply:
I rephrased the comment: I **propose** to change this part in a way above. So please implement it within this PR.

Reply:
Okay, I see. I will update the code . thanks.

Reply:
Done.

Reply:
@vladimir-paramuzov Could you please review it again?